### PR TITLE
feat(routing): build the gRPC interface for communicating with the external service to perform elimination routing

### PR DESCRIPTION
**Pull request by Aprabhat19**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
build the gRPC interface for communicating with the external service to perform elimination routing

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
- Communication is being established with dynamo
Screenshot 2024-12-05 at 6 03 54 PM

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by Chethan-rao — APPROVED** [review]

File: crates/external_services/src/grpc_client/dynamic_routing/elimination_rate_client.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    fn foreign_try_from(config: EliminationConfig) -> Result<Self, Self::Error> {
+        Ok(Self {
+            bucket_size: config
+                .bucket_size
+                .get_required_value("bucket_size")
+                .change_context(DynamicRoutingError::MissingRequiredField {
+                    field: "bucket_size".to_string(),
+                })?,
+            bucket_leak_interval_in_secs: config
+                .bucket_ttl_in_mins
```
Comment (Chethan-rao) [inline crates/external_services/src/grpc_client/dynamic_routing/elimination_rate_client.rs:151]: could u rename this too?

**Review by Chethan-rao — CHANGES_REQUESTED** [review]

**Comment by Chethan-rao** [comment]
please add test cases to all 3 endpoints and check the validations too

**Review by Chethan-rao — APPROVED** [review]

**Review by NishantJoshi00 — APPROVED** [review]

**Comment by NishantJoshi00** [comment]
> please add test cases to all 3 endpoints and check the validations too

+1 on that.

It would be better to have test cases. attached showing an example execution of these functionality.

**Comment by Aprabhat19** [comment]
> > please add test cases to all 3 endpoints and check the validations too
> 
> +1 on that.
> 
> It would be better to have test cases. attached showing an example execution of these functionality.

Okay sure

**Pull request merged**